Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,250 @@
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_Bz_lev" + std::to_string(lev) + " (T)";
+                ofs << m_sep;
+                ofs << "[" << c++ << "]probe_|B|_lev" + std::to_string(lev) + " (T)";
+            }
+            ofs << std::endl;
+            // close file
+            ofs.close();
+        }
+    }
+}
+// end constructor
+
+// function that computes maximum field values
+void FieldProbe::ComputeDiags (int step)
+{
+    // Judge if the diags should be done
+    if (!m_intervals.contains(step+1)) { return; }
+
+    // get a reference to WarpX instance
+    auto & warpx = WarpX::GetInstance();
+
+    // get number of level
+    const auto nLevel = warpx.finestLevel() + 1;
+
+    // loop over refinement levels
+    for (int lev = 0; lev < nLevel; ++lev)
+    {
+
+        const amrex::Geometry& gm = warpx.Geom(lev);
+        const auto prob_lo = gm.ProbLo();
+        const auto cell_size = gm.CellSizeArray();
+
+        const int i_probe = amrex::Math::floor((x_probe - prob_lo[0])/cell_size[0]);
+        const int j_probe = amrex::Math::floor((y_probe - prob_lo[1])/cell_size[1]);
+        const int k_probe = amrex::Math::floor((z_probe - prob_lo[2])/cell_size[2]);
+
+        // get MultiFab data at lev
+        const amrex::MultiFab & Ex = warpx.getEfield(lev,0);
+        const amrex::MultiFab & Ey = warpx.getEfield(lev,1);
+        const amrex::MultiFab & Ez = warpx.getEfield(lev,2);
+        const amrex::MultiFab & Bx = warpx.getBfield(lev,0);
+        const amrex::MultiFab & By = warpx.getBfield(lev,1);
+        const amrex::MultiFab & Bz = warpx.getBfield(lev,2);
+
+        constexpr int noutputs = 8; // probing Ex,Ey,Ez,|E|,Bx,By,Bz and |B|
+        constexpr int index_Ex = 0;
+        constexpr int index_Ey = 1;
+        constexpr int index_Ez = 2;
+        constexpr int index_absE = 3;
+        constexpr int index_Bx = 4;
+        constexpr int index_By = 5;
+        constexpr int index_Bz = 6;
+        constexpr int index_absB = 7;
+
+        // Prepare interpolation of field components to cell center
+        // The arrays below store the index type (staggering) of each MultiFab, with the third
+        // component set to zero in the two-dimensional case.
+        amrex::IndexType const Extype = Ex.ixType();
+        amrex::IndexType const Eytype = Ey.ixType();
+        amrex::IndexType const Eztype = Ez.ixType();
+        amrex::IndexType const Bxtype = Bx.ixType();
+        amrex::IndexType const Bytype = By.ixType();
+        amrex::IndexType const Bztype = Bz.ixType();
+
+        amrex::Real hv_Ex, hv_Ey, hv_Ez, hv_E, hv_Bx, hv_By, hv_Bz, hv_B;
+        int probe_proc = -1;
+        // MFIter loop to interpolate fields to cell center and get maximum values
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+        for (amrex::MFIter mfi(Ex, TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+            const amrex::Box& box = mfi.validbox();
```

Comment:
```suggestion
            const amrex::Box& box = amrex::enclosedCells(mfi.nodaltilebox());
```

I'm wondering if we should do this? Otherwise, I'm thinking that there may be a possibility, depending on the staggering, that the box contains the point `(i_probe, j_probe, k_probe)` although the position `(x_probe, y_probe, z_probe)` is in the guard cells. In which case, it's possible that there could be not enough points to perform the interpolation with order `WarpX::nox`.

Reply:
Sounds fair! I've changed as you suggested.